src/transformers/models/tvp/modeling_tvp.py

```diff
@@ -0,0 +1,914 @@
+        self.pad_left = nn.Parameter(
+            torch.randn([1, frame_num, 3, image_size - visual_prompt_size * 2, visual_prompt_size])
+        )
+        self.pad_right = nn.Parameter(
+            torch.randn([1, frame_num, 3, image_size - visual_prompt_size * 2, visual_prompt_size])
+        )
+
+    def forward(self, pixel_values):
+        if self.visual_prompter_apply != "add":
+            visual_prompt_mask = torch.ones([self.max_img_size, self.max_img_size], dtype=pixel_values.dtype)
+            pixel_values *= visual_prompt_mask
+        if self.visual_prompter_apply != "remove":
+            base = torch.zeros(1, self.frame_num, 3, self.base_size, self.base_size)
+            prompt = torch.cat([self.pad_left, base, self.pad_right], dim=4)
+            prompt = torch.cat([self.pad_up, prompt, self.pad_down], dim=3)
+            prompt = torch.cat(pixel_values.size(0) * [prompt])
+            pixel_values += prompt.to(pixel_values.dtype)
+        return pixel_values
+
+
+TVP_PROMPTER_CLASSES_MAPPING = {
+    "framedownpad": TvpFrameDownPadPrompter,
+    "framepad": TvpFramePadPrompter,
+}
+
+
+@add_start_docstrings(
+    "The bare Tvp Model transformer outputting BaseModelOutputWithPooling object without any specific head on" " top.",
+    TVP_START_DOCSTRING,
+)
+class TvpModel(TvpPreTrainedModel):
+    def __init__(self, config):
+        super(TvpModel, self).__init__(config)
+        self.config = config
+        self.vision_model = TvpVisionModel(config)
+        self.transformer = TvpTransformer(config)
+        if config.visual_prompter_type not in TVP_PROMPTER_CLASSES_MAPPING:
+            raise ValueError("`visual_prompter_type` must be in (framedownpad, framepad)")
+        else:
+            self.visual_prompter = TVP_PROMPTER_CLASSES_MAPPING[config.visual_prompter_type](
+                config.max_img_size,
+                config.visual_prompt_size,
+                config.num_frames,
+                config.max_img_size,
+                config.visual_prompter_apply,
+            )
+
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(TVP_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=BaseModelOutputWithPooling, config_class=TvpConfig)
+    def forward(
+        self,
+        input_ids: Optional[torch.LongTensor] = None,
+        pixel_values: Optional[torch.FloatTensor] = None,
+        attention_mask: Optional[torch.LongTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ):
+        r"""
+        Returns:
+
+        Examples:
+        ```python
+        >>> import torch
+        >>> from transformers import AutoConfig, AutoTokenizer, TvpModel
+
+        >>> config = AutoConfig.from_pretrained("Jiqing/tiny-random-tvp")
+        >>> model = TvpModel(config)
```

Comment:
Why create a model from randomly initialized weights here, rather than the checkpoint?